consensus/ethash/ethash.go

```diff
@@ -232,6 +262,42 @@ func newCache(epoch uint64, epochLength uint64) interface{} {
 	return &cache{epoch: epoch, epochLength: epochLength}
 }
 
+// isBadCache checks a given caches/datsets keccak256 hash against bad caches (ecip-1099)
+// this is incase the client has already written non-ecip1099 caches to disk,
+// instead of blindly trusting as seedhashes/filename match, compare checksums.
+func isBadCache(epoch uint64, epochLength uint64, data []uint32) (bool, string) {
+	// Check for bad caches/datasets at ecip-1099 transitions
+	if epochLength == epochLengthECIP1099 {
+		var badCache string
+		var badDataset string
+		var hash string
+
+		if epoch == 42 { // mordor
+			hash = uint32Array2Keccak256(data)
+			// bad cache generated using: geth makecache 2520001 [path] --epoch.length=30000
```

Comment:
I love these comments.

Reply:
How did you generate the keccak hash? In Go?

Reply:
Generated using a modified version of the current code. see the dev/dagchecksums branch, a little hacky but `geth makecache` and `geth makedag` will hash the chache/dag after generating and log.info it.

e.g 
```
$ ./build/bin/geth makecache 2520001 ~/.ethereum/mordor/geth/etchash/ --epoch.length=30000
INFO [10-21|18:23:29.712] Checksum'd new ethash dataset            epoch=84 path=/home/xocel/.ethereum/mordor/geth/etchash/cache-R23-002440e15267eebd hash=0xafa2a00911843b0a67314614e629d9e550ef74da4dca2215c475a0f93333aedc
```